batch/batch/driver/pricing.py

```diff
@@ -9,6 +9,7 @@ class Price(abc.ABC):
     effective_start_date: int
     effective_end_date: Optional[int]
     time_updated: int
+    sku: str
```

Comment:
Does this cause a `pyright` error if you don't add a `sku` attribute to any of the subclasses? AFAIK forgetting to set `sku` would only error at runtime. Instead of making these declarations and require subclasses to set `self.sku = sku` in every single subclass, can you make a constructor here in `Price` that sets these and the subclasses can call `super().__init__(region, effective_start_date, …)`? That way there's an easy statically caught error that you didn't call the constructor with the right number of arguments.

Side note: Do you know if these are really *instance attribute* declarations and not class attribute declarations? I tried playing around in `IPython` and it looks like by not setting them they don't appear as attributes of the class, but they're not required to be attributes of the instance either, so I'm not really sure what they're doing in this case.